libbeat/publisher/queue/spool/spool.go

```diff
@@ -0,0 +1,221 @@
+package spool
+
+import (
+	"os"
+	"sync"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common/atomic"
+	"github.com/elastic/beats/libbeat/publisher/queue"
+	"github.com/elastic/xbeats/txfile"
+	"github.com/elastic/xbeats/txfile/pq"
+)
+
+// Spool implements an on-disk queue.Queue.
+type Spool struct {
+	ctx *spoolCtx
+
+	// producer/input support
+	inBroker *inBroker
+
+	// consumer/output support
+	outBroker *outBroker
+
+	eventer queue.Eventer
+
+	queue *pq.Queue
+	file  *txfile.File
+}
+
+type spoolCtx struct {
+	logger logger
+	wg     sync.WaitGroup
+	active atomic.Bool
+	done   chan struct{}
+
+	handlers []func()
```

Comment:
I think calling the list of functions set by users to run when a `spoolCtx` is too general. I think handlers can be called whenever they are needed. But this list contains only a special kind of handlers which run during closing the session.
I would rather name it `onCloseCBs`, `onCloseCallbacks` or `closingCallbacks`. This change would make more obvious when these functions are called and what their role is.

Reply:
agreed ^

Reply:
Urgh... I felt dirty when introducing the `handlers`. No I'm feeling more "dirty" giving the impression we want multiple of these callbacks. Let's see if I can get rid of `handlers` or other callbacks.

Also not, these are for internal use only. No external lib/component is ever allowed to install a custom handler! The idea of `spoolCtx` is to provide a very small required subset of `context.Context` in order to deal with queue shutdown.

Reply:
Haha... handlers is not used anymore for quite some time -> removing it \o/